src/pudl/analysis/mcoe.py

```diff
@@ -474,40 +478,28 @@ def mcoe(
         .pipe(pudl.validate.no_null_cols, df_name="fuel_cost + capacity_factor")
     )
 
-    # Combine MCOE derived values with all the generator attributes:
-    mcoe_out = (
-        pd.merge(
-            left=(
-                pudl_out.gens_eia860()
-                .assign(year=lambda x: x.report_date.dt.year)
-                .drop("report_date", axis="columns")
-            ),
-            right=mcoe_out.assign(year=lambda x: x.report_date.dt.year),
-            # This "how" determines whether MCOE or gens_eia860 is the backbone
-            how="left" if all_gens else "right",
-            on=["year", "plant_id_eia", "generator_id"],
-        )
-        .astype({"year": str})
-        .assign(report_date=lambda x: x.report_date.fillna(pd.to_datetime(x.year)))
-        .drop("year", axis="columns")
-        .pipe(pudl.validate.no_null_rows, df_name="mcoe_all_gens", thresh=0.9)
-    )
+    # Combine MCOE derived values with generator attributes
+    if gens_cols == "all":
+        gens = pudl_out.gens_eia860()
+    elif gens_cols is None:
+        gens = pudl_out.gens_eia860()[DEFAULT_GENS_COLS]
+    else:
+        gens = pudl_out.gens_eia860()[list(set(DEFAULT_GENS_COLS + gens_cols))]
+
+    mcoe_out = pudl.helpers.full_timeseries_date_merge(
```

Comment:
Should there be a parameter that allows you to generate the MCOE table without the time series fill in and have this just be `date_merge`?

Reply:
I went ahead and added this parameter. Lmk if it doesn't make sense. It defaults to just do the `date_merge` not the `full_timeseries_date_merge`.